### PR TITLE
adding tests for files

### PR DESCRIPTION
**Pull request by mohamedasifs123**
#### What type of PR is this?
kind feature

#### What this PR does / why we need it:

adding tests for files
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
none
```

### CODE REVIEW

File: pkg/utils/file/compress_test.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	"bytes"
+	"compress/gzip"
+	"io"
+	"testing"
+)
+
+func TestCompress(t *testing.T) {
+	tests := []struct {
+		name     string
+		ext      string
```
Comment (wzshiming) [inline pkg/utils/file/compress_test.go:26]:
```suggestion
		ext      string
		data     []byte
```

Don't hard-coded the data

File: pkg/utils/file/compress_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		} else if buf.String() != "hello world" {
+			t.Errorf("expected 'hello world', got %s", buf.String())
+		}
+	}
+}
+
+func TestDecompress(t *testing.T) {
+	tests := []struct {
+		name       string
+		compressed bool
```
Comment (wzshiming) [inline pkg/utils/file/compress_test.go:87]:
```suggestion
        data       []byte
		compressed bool
```

ditto

File: pkg/utils/file/compress_test.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+		{
+
+			name:       "test.tgz",
+			compressed: true,
+		},
+		{
+
+			name:       "test.zip",
+			compressed: false,
+		},
```
Comment (wzshiming) [inline pkg/utils/file/compress_test.go:107]: add a case with gzip compressed data

File: pkg/utils/file/download_test.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+	"os"
+	"testing"
+
+	"sigs.k8s.io/kwok/pkg/utils/path"
+)
+
+func TestDownloadWithCacheAndExtract(t *testing.T) {
+	ctx := context.Background()
+	cacheDir := "/tmp/cache"
+	src := "http://example.com/file.tar.gz"
```
Comment (wzshiming) [inline pkg/utils/file/download_test.go:31]: Don't depend on an external source, start a test server instead.

**Review by wzshiming — COMMENTED** [review]
/ok-to-test

File: pkg/utils/file/download_test.go (hunk 4 above)
Comment (mohamedasifs123) [reply pkg/utils/file/download_test.go:31]: In test server where can I place the files?

File: pkg/utils/file/compress_test.go (hunk 1 above)
Comment (mohamedasifs123) [reply pkg/utils/file/compress_test.go:26]: Shall I need to have separate data for all test cases?

File: pkg/utils/file/download_test.go (hunk 4 above)
Comment (wzshiming) [reply pkg/utils/file/download_test.go:31]: import "net/http/httptest" and use `httptest.NewServer()` to create a test server, and use `gzip` to compress the content as http response, and put the output in the tmp dir.

File: pkg/utils/file/compress_test.go (hunk 3 above)
Comment (mohamedasifs123) [reply pkg/utils/file/compress_test.go:119]: I think The decompress func only support .gz iand .tgz .

**Pull request closed without merging**